include/zarray/zarray_impl.hpp

```diff
@@ -17,9 +17,85 @@
 #include <xtensor/xnoalias.hpp>
 #include <xtensor/xscalar.hpp>
 #include <xtensor/xshape.hpp>
+#include <xtensor/xshape.hpp>
+#include "xtensor/xstrided_view.hpp"
 
 namespace xt
 {
+    template <class CTE>
+    class zarray_wrapper;
+
+    template <class CTE>
+    class zchunked_wrapper;
+
+    template <class CTE>
+    class zexpression_wrapper;
+
+    namespace detail
+    {
+        template <class E>
+        struct is_a_strided_view : std::false_type
+        {
+        };
+
+        template <class E, class S>
```

Comment:
I think you could add the [additional template arguments](https://github.com/xtensor-stack/xtensor/blob/master/include/xtensor/xstrided_view.hpp#L125) of `xstrided_view` to be sure we can match all the cases.